backend/prisma/schema.prisma

```diff
@@ -8,11 +8,11 @@ datasource db {
 }
 
 model User {
-  id   String @id @default(uuid())
-  email   String   @unique
-  firstName    String 
-  lastName String 
-  password String 
-  birthDate DateTime? 
+  id        String    @id @default(uuid())
+  email     String?    @unique
```

Comment:
unfortunately since the prev migration that you deleted has been run by other members, we will begin running into migration issues. however, given that it is early in the project, we have 2 options

1) hacky. have everyone drop their db's and run a new migration that creates the User table based on the newly modified schema you have made here

2) best practice. write a new migration that handles altering the table schema to transition from the old User table schema to this new one. i _believe_ prisma will handle most of this logic for you but you'll need to figure out how that part works (and hopefully pass the knowledge on to your teammates!)

Reply:
Ok, I will research more into schema migration from old to new.

Reply:
Fortunately, I altered column email, password, birthDate to nullable and added nullable column googleID, these changes to the schema will not cause our database to reset.

After researching, I found a method to migrate schema without data loss. When prompted a database reset, we can use the prisma migrate --create-only command to create a .sql file with the new schema from schema.prisma file, this will only create a new migration but does apply the changes. Then, we can edit the .sql file to alter columns: drop column, set default values, set non-null constraints etc. Finally, after making the necessary changes to the .sql file, we can just use prisma migrate dev to apply the custom migration to our database. As long as our database end state matches with schema.prisma, we can migrate without data loss.